### PR TITLE
Add tests

### PR DESCRIPTION
**Pull request by sarala**
I have written a couple of test for controller:providers.show.organizations.new

There are few things that I don't understand.

1) how do I create an organisation? I created a factory but make('organisation') doesn't seem to create the object.
2) searching for organisation type using the full name doesn't seem to work. For example, searching for nationalInstitution returns 0.

### CODE REVIEW

**Comment by mfenner** [comment]
When you look at the available models, you see that there is no `organization`, as this is a subtype (memberType: consortium_organization) of `provider`.

**Comment by mfenner** [comment]
The search for organization type reformats the values in the predefined list: https://github.com/datacite/bracco/blob/master/app/templates/providers/new.hbs#L68

You need to search for `National Institution`.

File: tests/factories/organization.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import FactoryGuy from 'ember-data-factory-guy';
+
+FactoryGuy.define('provider', {
```
Comment (richardhallett) [inline tests/factories/organization.js]: This should be "organization" then you'll be able to make("organization") in the tests

File: tests/unit/controllers/providers-test.js [hunk 2]
```diff
@@ -1,12 +1,54 @@
...
+    let controller = this.owner.lookup('controller:providers.show.organizations.new');
+
+    let model = {
+      'provider': make('provider'),
+      'organization': make('provider', { memberType: 'consortium_organization' }),
+    };
+    controller.set('model', model);
+    controller.send('selectOrganizationType', 'nationalInstitution');
+    assert.equal(controller.model.organization.get('organizationType'), 'nationalInstitution');
+    controller.send('searchOrganizationType', 'National Institution');
```
Comment (richardhallett) [inline tests/unit/controllers/providers-test.js]: This doesn't search the friendly name, it's actually looking through the types defined in the controller which are lie "nationalInstitution" so "national" would work.

File: tests/unit/controllers/dois-test.js [hunk 3]
```diff
@@ -1,12 +1,29 @@
...
+  // setupFactoryGuy(hooks);
 
   // Replace this with your real tests.
   test('it exists', function(assert) {
     let controller = this.owner.lookup('controller:dois');
     assert.ok(controller);
   });
+
+  // promise error needs to be fixed
+  /*
```
Comment (richardhallett) [inline tests/unit/controllers/dois-test.js]: As I also looked into this, the problem is when the factory creates into the store, the submit is trying to send this data to the real API and therefore causes an error on that side.
I'm not sure what is best here, but we might need to be able to mock the API service so it responds as we expect.

File: tests/factories/organization.js (hunk 1 above)
Comment (sarala) [reply tests/factories/organization.js]: No, can't use organisation as there is no organisation model.

Comment (richardhallett) [reply tests/factories/organization.js]: You're right, my mistake I misunderstod what this was doing.
Maybe you can use polymorphic model.
https://github.com/danielspaniel/ember-data-factory-guy#polymorphic-model

**Comment by sarala** [comment]
I am still struggling here. I don't know how to test for consortiums and ror ids. Could you please have a look how to fix them.

**Pull request merged**